### ISSUE TITLE
Yikes error generated when opening empty session

### ISSUE THREAD

**Issue by highdownts** [RickStrahl/WestWindWebSurge#83]
The following between the WebSurge options was the original file content. Cannot open this file without an error.

When I do capture data and save, the WebSurge options are not saved. The GUI options revert to the defaults.

I installed the latest Build 1.20 64bit.

----- Start WebSurge Options -----

{
  "SiteBaseUrl": null,
  "ReplaceDomain": null,
  "NoProgressEvents": true,
  "DelayTimeMs": 0,
  "NoContentDecompression": true,
  "CaptureMinimalResponseData": true,
  "MaxResponseSize": 0,
  "ReplaceQueryStringValuePairs": null,
  "ReplaceCookieValue": null,
  "ReplaceAuthorization": null,
  "Username": null,
  "Password": null,
  "Users": [],
  "RequestTimeoutMs": 15000,
  "RandomizeRequests": false,
  "WarmupSeconds": 2,
  "IgnoreCertificateErrors": false,
  "TrackPerSessionCookies": true,
  "ReloadTemplates": false,
  "FormattedPreviewTheme": "vscodelight",
  "LastThreads": 5,
  "LastSecondsToRun": 10
}

// This file was generated by West Wind WebSurge
// Get your free copy at https://websurge.west-wind.com
// to easily test or load test the requests in this file.

----- End WebSurge Options -----

Here is the error condition:

4/29/2021 9:07:23 AM - Object reference not set to an instance of an object.
---
mscorlib
   at System.Object.GetType()
   at System.Windows.Forms.DataObject.DataStore.SetData(Object data)
   at System.Windows.Forms.Control.DoDragDrop(Object data, DragDropEffects allowedEffects)
   at System.Windows.Forms.Control.OnMouseMove(MouseEventArgs e)
   at System.Windows.Forms.Control.WmMouseMove(Message& m)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ListView.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)

None of my existing session files will save the options. The files do not contain the options header as shown above. I have tried manually adding, but the option values are not read on opening the session.

Regards, Warren

**Comment by highdownts** [comment]
I went to my backup files and was able to open an older file. The WebSurge option order is different on the backup file relative to the current option list. I save the file that worked under a different name and did a diff of the two files. The only difference is the option list as shown below. There is one new parameter and the order is different. When I replaced the new options with the older format, the new file opens and runs properly.

Backup File
----- Start WebSurge Options -----

{
					  
						
  "NoProgressEvents": true,
  "DelayTimeMs": 0,
  "NoContentDecompression": true,
  "CaptureMinimalResponseData": true,
  "MaxResponseSize": 0,
  "ReplaceQueryStringValuePairs": null,
  "ReplaceDomain": null,
  "ReplaceCookieValue": null,
  "ReplaceAuthorization": null,
  "Username": null,
  "Password": null,
  "Users": [],
  "RandomizeRequests": false,
  "RequestTimeoutMs": 15000,
							 
  "WarmupSeconds": 2,
								   
								 
  "ReloadTemplates": false,
  "FormattedPreviewTheme": "vscodelight",
  "LastThreads": 5,
  "IgnoreCertificateErrors": false,
  "TrackPerSessionCookies": true,
  "LastSecondsToRun": 20
}

// This file was generated by West Wind WebSurge
// Get your free copy at https://websurge.west-wind.com
// to easily test or load test the requests in this file.

----- End WebSurge Options -----

New Saved File
----- Start WebSurge Options -----

{
  "SiteBaseUrl": null,
  "ReplaceDomain": null,
  "NoProgressEvents": true,
  "DelayTimeMs": 0,
  "NoContentDecompression": true,
  "CaptureMinimalResponseData": true,
  "MaxResponseSize": 0,
  "ReplaceQueryStringValuePairs": null,
						
  "ReplaceCookieValue": null,
  "ReplaceAuthorization": null,
  "Username": null,
  "Password": null,
  "Users": [],
							 
  "RequestTimeoutMs": 15000,
  "RandomizeRequests": false,
  "WarmupSeconds": 2,
  "IgnoreCertificateErrors": false,
  "TrackPerSessionCookies": true,
  "ReloadTemplates": false,
  "FormattedPreviewTheme": "vscodelight",
  "LastThreads": 5,
								   
								 
  "LastSecondsToRun": 20
}

// This file was generated by West Wind WebSurge
// Get your free copy at https://websurge.west-wind.com
// to easily test or load test the requests in this file.

----- End WebSurge Options -----

**Comment by highdownts** [comment]
The following are the files downloaded during the site test. Htm and JavaScript files are zipped.

**URL										Body**
/index.htm                                                                       23,733
/0/humphries-antiques.jpg				16,122
/0/instantssl-trust-seal-md-159x42.png	3,031
/0/ford-tough.jpg						10,956
/app/bundle.min.js						115,570
/app/deferredBundle.min.js				263,240
/images/favicon.ico						9,812
								  Total 442,464

WebSurge Options to reduce processing load:
  "NoProgressEvents": true,
  "NoContentDecompression": true,
  "CaptureMinimalResponseData": true,
  "MaxResponseSize": 10000,
  "LastSecondsToRun": 120
  
In Task Manager, WebSurge memory usage got up to a little more than 1 Gbytes.

At the end of the 120 second run, a Yikes dialog was displayed and the error log contained the following:

4/29/2021 10:51:28 AM - Arithmetic operation resulted in an overflow.
---
System.Core
   at System.Linq.Enumerable.Sum(IEnumerable`1 source)
   at WebSurge.ResultsParser.<>c__DisplayClass4_0.<UrlSummary>b__2(String key, IEnumerable`1 uls)
   at System.Linq.Lookup`2.<ApplyResultSelector>d__13`1.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at WebSurge.ResultsParser.GetResultReport(RequestWriter requestWriter, Int32 totalTimeTakenMs, Int32 threadCount)
   at WebSurge.ResultsParser.GetResultReportHtml(RequestWriter writer, Int32 totalTimeTakenMs, Int32 threadCount)
   at WebSurge.StressTestForm.ParseResults(List`1 results)

The requests per second were about 1,600 and the graph of the results displayed the whole test run.

I am not sure what has changed, but when I first started using the product, I could run for 10 minutes or more without failure.

Am I missing something in the setup?

I was attempting to create a test project when I encountered the issue documented in this bug report.

I will modify the text file manually to preserve the old option format.

Regards,

Warren
